src/patch/parse.ts

```diff
@@ -30,47 +31,67 @@ export function parse(textline: string): Patch[] {
     if (m[2] === '') {
       patch.start1--
       patch.length1 = 1
+      patch.byteLength1 = 1
     } else if (m[2] === '0') {
       patch.length1 = 0
+      patch.byteLength1 = 0
     } else {
       patch.start1--
       patch.length1 = toInt(m[2])
+      patch.byteLength1 = patch.length1
     }
 
     if (m[4] === '') {
       patch.start2--
       patch.length2 = 1
+      patch.byteLength2 = 1
     } else if (m[4] === '0') {
       patch.length2 = 0
+      patch.byteLength2 = 0
     } else {
       patch.start2--
       patch.length2 = toInt(m[4])
+      patch.byteLength2 = patch.length2
```

Comment:
```suggestion
      // The patch itself will contain the UTF-8 length:
      patch.byteLength2 = toInt(m[4])
      // We start with length2 being set with the same, but then we adjust for it later:
      patch.length2 = patch.byteLength2
```

Flip this around and clarify a bit that `length2` is being mutated?